### PR TITLE
feat(fs): improve `readTextFile` and `readTextFileLines` performance

### PR DESCRIPTION
**Pull request by amrbashir**

### CODE REVIEW

File: plugins/fs/src/commands.rs [hunk 1]
```diff
@@ -15,8 +15,8 @@ use tauri::{
 use std::{
     borrow::Cow,
     fs::File,
-    io::{BufReader, Lines, Read, Write},
-    path::PathBuf,
+    io::{BufRead, BufReader, Read, Write},
+    path::{Path, PathBuf},
```
Comment (amrbashir) [inline plugins/fs/src/commands.rs]:
```suggestion
    path::PathBuf,
```

File: plugins/fs/guest-js/index.ts [hunk 2]
```diff
@@ -811,19 +816,35 @@ async function readTextFileLines(
...
 
-      // an iteration is over, reset rid for next iteration
-      if (done) this.rid = null
+      const bytes =
+        arr instanceof ArrayBuffer ? new Uint8Array(arr) : Uint8Array.from(arr)
+
+      // Rust side will never return an empty array for this command and
+      // ensure there is at least one elements there.
+      //
+      // This is an optimization to include wether we finished iteration or not (1 or 0)
```
Comment (FabianLars) [inline plugins/fs/guest-js/index.ts]:
```suggestion
      // This is an optimization to include whether we finished iteration or not (1 or 0)
```

**Review by FabianLars — DISMISSED** [review]
(tested)

**Review by FabianLars — DISMISSED** [review]

**Comment by FabianLars** [comment]
ngl i hate this dismissal setting so much

**Review by FabianLars — APPROVED** [review]

**Pull request merged**